src/main/java/org/jnode/fs/xfs/attribute/XfsAttribute.java

```diff
@@ -0,0 +1,146 @@
+package org.jnode.fs.xfs.attribute;
+
+import org.jnode.fs.xfs.XfsObject;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A XFS SELinux Attribute.
+ *
+ * @author
+ */
+public class XfsAttribute extends XfsObject {
+
+    /**
+     * The value of the flags.
+     */
+    private final int flags;
+
+    /**
+     * The length of the name.
+     */
+    private final int nameLength;
+
+    /**
+     * The length of the attribute.
+     */
+    private final int valueLength;
+
+    /**
+     * The name of the attribute.
+     */
+    private final String attributeName;
+
+    /**
+     * The value of the attribute.
+     */
+    private final String attributeValue;
+
+    /**
+     * Creates an attribute instance.
+     *
+     * @param data of the inode.
+     * @param offset of the inode's data
+     * @throws IOException if an error occurs reading in the super block.
+     */
+    public XfsAttribute(byte[] data, long offset) throws IOException {
+        super(data, (int) offset);
+        nameLength = (int) read(0, 1);
+        valueLength = (int) read(1, 1);
+        flags = (int) read(2, 1);
+        ByteBuffer buffer = ByteBuffer.allocate(nameLength + valueLength);
+        System.arraycopy(data, (int) offset + 3, buffer.array(), 0, (nameLength + valueLength));
+        final String nameval = new String(buffer.array(), StandardCharsets.US_ASCII);
+        attributeName = nameval.substring(0, nameLength);
+        attributeValue = nameval.substring(nameLength).replaceAll("\0", "");
```

Comment:
Looks like you can do something similar in `Superblock.getName()`.

Reply:
yes, thanks I applied the same instructions in the superblock class. thanks.